NewTestDochkin

Dochkin-sergey opened the pull request.

Dochkin-sergey committed:
NewTestDochkin

Dochkin-sergey committed:
NewTestDochkin1

Dochkin-sergey committed:
NewTestDochkin

vladimirStest commented on .idea/misc.xml:
```diff
@@ -11,5 +11,5 @@
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" default="true" project-jdk-name="openjdk-21" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="openjdk-21" project-jdk-type="JavaSDK" />
```
Правки pom файла коммитить и пушить не нужно.

vladimirStest commented on allure-results/062379a4-02ca-4a27-8de5-8a7ebb774bca-container.json:
```diff
@@ -0,0 +1 @@
+{"uuid":"062379a4-02ca-4a27-8de5-8a7ebb774bca","name":"Проверка, что кнопка Get started free активна","children":["a3c27d5d-7490-4991-a503-588678979cf4"],"befores":[],"afters":[],"start":1742233734677,"stop":1742233741872}
```
Каталог allure-results и файлы из него коммитить и пушить не нужно. Как сейчас удалить: удалить каталог у себя локально, закоммитить и запушить правки после удаления. В будущем НЕ отмечать галочкой при коммите пуше этот каталог и файлы

Dochkin-sergey replied:
Исправлю в следующем комите

Dochkin-sergey committed:
NewTestDochkinPass

Dochkin-sergey committed:
Исправлены названия классов.
Исправлен локатор по коментарию. Добавлено логирование

Dochkin-sergey committed:
Убрано логирование SLF4J
Добавлена аннотация Step

vladimirStest commented on .gitignore:
```diff
@@ -30,7 +30,7 @@ target/
 !**/src/main/**/build/
 !**/src/test/**/build/
-
+pom.xml
```
pom.xml в `.gitignore` добавлять не нужно

vladimirStest commented on src/test/java/com/example/jetbrainstest/pages/TeamcityPage.java:
```diff
@@ -4,26 +4,29 @@
+    private final Logger LOG = LoggerFactory.getLogger(TeamcityPage.class);
 
-    @FindBy(xpath = "//*[contains(text(), 'Get started free')]")
+    @FindBy(xpath = "//div//div/a/span[contains(text(), 'Get started free')]")
```
В данном случае локатор опять получился нехороший:
```
1) div уникально не описывает элементы зачастую, лучше брать по class или id, например
2) Не нужно много родителей добавлять в путь, чаще всего достаточно одного, но уникального
```
Неидеальным (из-за не совсем уникальности), но приемлемым вариантом будет `xpath = "//a/span[contains(text(), 'Get started free')]"`

vladimirStest commented on src/test/java/com/example/jetbrainstest/pages/TeamcityPage.java:
```diff
@@ -16,13 +17,13 @@ public class TeamcityPage {
 
+    @Step ("Проверка видимости текста GetStartedFree")
     public boolean buttonActivityGetStartedFree() {
-        LOG.info("Проверка видимости текста GetStartedFree");
```
Логирование требуется в проекте, даже если есть аллюр, чтобы смотреть информацию по запуску в среде разработки

vladimirStest commented on src/test/java/com/example/jetbrainstest/pages/TeamcityPage.java:
```diff
@@ -16,13 +17,13 @@ public class TeamcityPage {
     @FindBy(css = "[data-test=\"site-header-profile-action\"]")
     private WebElement AuthorizationIcon;
 
+    @Step ("Проверка видимости текста GetStartedFree")
```
ctrl+alt+l для форматирования кода, пробел после @Step лишний

Dochkin-sergey committed:
Поправил локатор
Привязал allure к отдельному классу
Кстати почему то перестал отображаться отчёт, пишет страница недоступна

Dochkin-sergey committed:
Добавил изменения по уроку
